assets/js/components/gateways/GatewayRegister.jsx

```diff
@@ -32,9 +32,27 @@ class GatewayRegister extends Component {
   constructor(props) {
     super(props)
 
+    this.state = {
+      showQRCode: props.gateway.public_key == null
+    }
+
     this.renderStep2 = this.renderStep2.bind(this)
   }
 
+  componentDidMount() {
+    const { gateway, socket, currentTeamId } = this.props
+    const channelName = `gateway:${gateway.id}`
+
+    let channel = socket.channel(channelName, {team_id: currentTeamId})
+
+    channel.join()
+      .receive("ok", resp => { console.log(`Joined ${channelName} successfully`, resp) })
+      .receive("error", resp => { console.log(`Unable to join ${channelName}`, resp) })
+
+    channel.on("registering", () => { this.setState({showQRCode: false}) })
+    channel.on("verified", () => { window.location.reload() })
```

Comment:
why do we need to force a reload when the status changes to verified?

Reply:
this allows gateway show page to repull the graphql data, otherwise, it doesn't show the verified gateway and stay on the loading status